### ISSUE TITLE
Doxygen Regression when documenting enums for C#(1.94 -> 1.9.5)

### ISSUE THREAD

**Issue by StefanBattmer** [doxygen/doxygen#9722]
Starting with Doxygen 1.9.5 enum members for a C# API are no longer documented. The corresponding HTML output just seems to omit the numerical constants for the enums.

**Expected behavior**
What is seen in the left half of the screenshot.

**Screenshots**
See attached Screenshot. For C++/C this is still working fine. Doxyfiles for C++ and C# are almost identical

![doxygenRegression](https://user-images.githubusercontent.com/33477553/207640786-e6ee6fc0-175f-48af-94fa-df8ba023df6f.PNG)

**To Reproduce**
Could provide a sample if that doesn't ring a bell, but haven't done yet

**Version**
1.9.4 is still working as expected, 1.9.5 isn't.

**Stack trace**
-

**Additional context**
Will be happy to provide more info as needed! After all I really like the tool! :-)

**Comment by albert-github** [comment]
> Could provide a sample if that doesn't ring a bell, but haven't done yet

No it doesn't ring a bell yet.
- when possible also test it against the current master.
- yes please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by StefanBattmer** [comment]
Hi,

we first build Doxygen from the master (about a week ago) more or less by accident. When we saw the problem we went back to the tagged 1.9.5 version and then finally to the tagged 1.9.4 version where the problem didn't occur. I prepared a minimal example with HTML output from the 1.9.4 and 1.9.5 versions. I think the main difference causing this can be found in the file named **namespacemv_1_1impact_1_1acquire.html**

All the best,

Stefan
[DoxygenIssue9722.zip](https://github.com/doxygen/doxygen/files/10238896/DoxygenIssue9722.zip)

**Comment by albert-github** [comment]
Problem is introduced in version 1.9.5, to be more precise the last good commit was:
```
Commit: a1ccbc3d009af87f874765ef33af2c02cbeb56cc [a1ccbc3]
Date: Sunday, June 5, 2022 9:01:37 PM

Fixed compilation problem in doxyapp.cpp
```

The first bad commit (i.e. next commit in master) is:
```
Commit: 35711f2d14acc81cff83f81f04c566ad8c8c6fe8 [35711f2]
Date: Monday, June 6, 2022 7:31:37 PM

Fix regressions introduced by changing the way references are resolved

- grouped members where stored in the tag file without scope
- strong enum values where not linkable
```

**Comment by StefanBattmer** [comment]
Great news! Thanks! Whenever we shall do some testing just reach out!

**Comment by doxygen** [comment]
@StefanBattmer Please test with the referenced commit if possible.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by StefanBattmer** [comment]
Thanks! The issue has been resolved!
